File: Modul 6/Soal 4/PRAK604-NIM-Nama.c
#include <stdio.h>
#include <string.h>

int main(){
    char code[100], message[100];
    int star = 0, hash = 0;

    fgets(code, sizeof(code), stdin);
    fgets(message, sizeof(message), stdin);

    code[strcspn(code, "\n")] = '\0';
    message[strcspn(message, "\n")] = '\0';

    if(strlen(code) != strlen(message)){
        printf("Panjang Kalimat berbeda, pesan palsu");
        return 0;
    }

    int panjang = strlen(code);
    for(int i = 0; i < panjang; i++){
        if(code[i] == ' ' && message[i] == ' '){
            continue;
        }

        if(code[i] == message[i]){
            printf("*");
            star++;
        }

        else{
            printf("#");
            hash++;
        }
    }

    printf("\n* = %d", star);
    printf("\n# = %d \n", hash);

    if(star >= hash){
        printf("Pesan Asli \n");
    }
    
    else{
        printf("Pesan Palsu \n");
    }
}

File: Modul 6/Soal 4/PRAK604-NIM-Nama.py
code = input()
message = input()

if len(code) != len(message):
    print("Panjang Kalimat berbeda, pesan palsu")
else:
    star = 0
    hash = 0
    result = ""

for i, j in zip(code, message):
    if i == " " and j == " ":
        continue

    if i == j:
        result += "*"
        star += 1
    else:
        result += "#"
        hash += 1

print(result)
print(f"* = {star}")
print(f"# = {hash}")

if star >= hash:
    print("Pesan Asli")
else:
    print("Pesan Palsu")